lib/fluent/plugin/in_tail.rb

```diff
@@ -490,9 +490,11 @@ def update_watcher(target_info, pe)
 
         if new_position_entry.read_inode == 0
           @tails[new_target_info] = setup_watcher(new_target_info, new_position_entry)
+          @tails[new_target_info].on_notify
```

Comment:
Constructed watchers should be registered ASAP, before running `on_notify`.
Otherwise shutting down process can't recognize the watcher so that it won't be killed.
This is the why the error I mentioned at https://github.com/fluent/fluentd/issues/3323#issuecomment-830634918 is occurred.